crates/lumina_shared/src/player/spawn_point.rs

```diff
@@ -10,43 +10,50 @@ impl Plugin for SpawnPointPlugin {
     fn build(&self, app: &mut App) {
         app.add_systems(
             PostUpdate,
-            init_spaceship.after(TransformSystem::TransformPropagate),
+            init_spaceships_at_spawn_points.after(TransformSystem::TransformPropagate),
         )
         .observe(on_add_spawned)
         .observe(on_remove_spawned);
     }
 }
 
-// Initialize spaceships and assign them alternately to teams.
-fn init_spaceship(
+/// Initialize spaceships and assign them to available spawn points.
+fn init_spaceships_at_spawn_points(
     mut commands: Commands,
-    q_spawn_points: Query<(&GlobalTransform, Entity), (With<SpawnPoint>, Without<SpawnPointUsed>)>,
-    mut q_spaceships: Query<
+    spawn_point_query: Query<
+        (&GlobalTransform, &TeamType, Entity),
+        (With<SpawnPoint>, Without<SpawnPointUsed>),
+    >,
+    mut spaceship_query: Query<
         (&mut Position, &mut Rotation, Entity),
-        (
-            With<Spaceship>,
-            With<SourceEntity>,
-            Without<SpawnPointEntity>,
-        ),
+        (With<Spaceship>, Without<SpawnPointEntity>),
     >,
 ) {
-    let mut spawn_points = q_spawn_points.iter();
-
-    for (mut position, mut rotation, entity) in q_spaceships.iter_mut() {
-        // Obtain a new spawn transform.
-        let Some((spawn_transform, spawn_entity)) = spawn_points.next() else {
-            return;
+    let mut available_spawn_points = spawn_point_query.iter();
```

Comment:
```suggestion
    let mut spawn_points = spawn_point_query.iter();
```

Too lengthy (nit pick)